pkg/client/client.go

```diff
@@ -0,0 +1,92 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/scheme"
+	cr "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	applicationv1alpha1 "github.com/giantswarm/apiextensions-application/api/v1alpha1"
+)
+
+// Client extends the client from controller-runtime
+type Client struct {
+	cr.Client
+}
+
+// New creates a new Kubernetes client for the provided kubeconfig file
+//
+// The client is an extension of the client from controller-runtime and provides some additional helper functions.
+// The creation of the client doesn't confirm connectivity to the cluster and REST discovery is set to lazy discovery.
```

Comment:
Its because it fails to create the client if it can't connect to the cluster which it can't do when it's still being created. I can update the comment.